File: src/decoders/news-survey.ts
import type { NewsCategory, SessionHandle, NewsSurvey } from "~/models";
import { decodeNewsQuestion } from "./news-question";
import { decodeNewsItem } from "./news-item";

export const decodeNewsSurvey = (survey: any, session: SessionHandle, categories: NewsCategory[]): NewsSurvey => {
  return {
    is: "survey",
    ...decodeNewsItem(survey, categories),

    questions: survey.listeQuestions.V.map((question: any) => decodeNewsQuestion(question, session)),
    isAnonymous: survey.reponseAnonyme
  };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/news-survey.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.NewsCategory
import ink.literate.pawnote.models.NewsSurvey
import ink.literate.pawnote.models.SessionInformation

import kotlinx.serialization.json.*

fun decodeNewsSurvey (survey: JsonObject, sessionInfo: SessionInformation, categories: List<NewsCategory>): NewsSurvey {
    return NewsSurvey(
        questions = survey["listeQuestions"]!!.jsonObject["V"]!!.jsonArray.map { decodeNewsQuestion(it.jsonObject, sessionInfo) },
        isAnonymous = survey["reponseAnonyme"]!!.jsonPrimitive.boolean
    )
}